strawberryfields/ops.py

```diff
@@ -2414,7 +2446,44 @@ def _decompose(self, reg, **kwargs):
 
         cmds = []
 
-        if not self.identity or not drop_identity:
+        if mesh == "rectangular_compact":
```

Comment:
With the addition of this code block, it looks like the linter is raising two issues:

https://www.codefactor.io/repository/github/xanaduai/strawberryfields/pull/584

'too many statements' and 'too many branches' is usually a sign that a method or function is trying to do too much, and it might be worth considering splitting it up into smaller functions, e.g., `_rectangular_compact_commands()` and just calling it here,
```python
cmds = _rectangular_compact_commands(...)
```

However, this is just a suggestion; to disable this warning, you can add the comment
```python
#  pylint: disable=too-many-statements,too-many-branches
```
at the very beginning of the function.

Reply:
I like the idea of moving this into a separate function. Just to be clear, do you think this is better as a standalone function rather than a method of the Interferometer class?

Reply:
It does have advantages, in that you can test `_rectangular_compact_commands()` (or whatever you name it 😆) independently of the rest of the `Interferometer` class. However, I will leave the decision up to you and Nico. Feel free to also leave it as is, you will just need to let pylint know to ignore those warnings.

Reply:
I've done this, thanks for the suggestion